### DISCUSSION TITLE
How to use function from typescript file in custom `server.js`

### DISCUSSION THREAD

**Discussion by shahidcodes** [vercel/next.js#24863] [Help] [answered]
I've a next app which uses typescript, I need to use custom server in order to do some things. I created a `server.js` file in root and everything works. But I don't know how to use some functions written in typescript file in `server.js` since it does not go through babel.

What's the way to do it?
Basically, I want to use a function exported from `bootstrap.ts` in my `server.js`
![image](https://user-images.githubusercontent.com/17620737/117294309-ed4f0c80-ae8b-11eb-9ca7-91ad82bb07a3.png)

**Comment by bobarros** [comment, marked as answer]
it's possible to convert server.js to server.ts or it's going to be too much trouble?

Because if it is simple, it will work by just importing the function inside server.ts

```javascript
import express from 'express';
import function from 'otherfile';
```

**Reply by shahidcodes** [reply]
Don't think I can just rename the server to .ts as doc says it doesn't go through babel or webpack and the way to start next app when using server is to run server.js file through node.js directly. 

https://nextjs.org/docs/advanced-features

**Reply by bobarros** [reply]
Most of the time you can't just rename a js file to a ts one. That's why I wrote convert.
Your link doesn't exist. It was suppose to be this one? https://nextjs.org/docs/advanced-features/custom-server

Anyway, I was asking if something like so was easier:

> https://github.com/vercel/next.js/tree/canary/examples/custom-server-typescript

Because all your project would stay with typescript and seems more like it should be.
You can test yourself using:

```
[npx, yarn etc] create next-app --example custom-server-typescript custom-server-typescript-app
```

**Reply by shahidcodes** [reply]
I misunderstood, my bad. That'll work. 

Thanks
